### ISSUE TITLE
Placeholders in "Last Activity" appear instead of real values

### ISSUE THREAD

**Issue by SafaAlfulaij** [translate/pootle#4121]
In Mozilla, I found this:
![Last Activity placeholders](http://i.imgur.com/XJsI2I2.png)

**Comment by unho** [comment]
This might be related to 3b74a95b.

**Comment by julen** [comment]
Thanks for the pointer @unho. That commit indeed introduced a regression: it removed template strings (those wrapped around backticks), which do parameter interpolation for free. If using regular strings, the placeholder replacement has to be done via the cumbersome `interpolate()` function.

I'd strongly favor fixing the string extraction methods to be able to seamlessly use template strings.

**Comment by dwaynebailey** [comment]
We're in a pickle here then as I can't find anything to extract ECMAScript template strings.  Not even any custom xgettext extractors.

You don't seem to have any control in xgettext over the quote marks.  But I managed to get the extractor to work if the string is ``gettext(`"Translatable text ${var}"`)`` @julen does that work in code?

**Comment by julen** [comment]
No it doesn't, that'd be the same as `'"foo bar"'` or `'"foo bar"'`, i.e. quotes end up being part of the actual string, which is not desired.

**Comment by dwaynebailey** [comment]
@julen yeah just tested here http://mozilla.locamotion.org/user/khaled/ and I get the stray quotes.

Do you have any pointers to an extraction tool?

**Comment by dwaynebailey** [comment]
Deployed a workaround on mlo but reverting 3b74a95

This needs ``interpolate`` or a proper extraction tool for templates.  Or someone to figure out how xgettext can be made to see backtick surrounded strings as extractable strings.

**Comment by dwaynebailey** [comment]
@julen, nope its still a messup.  I'm afraid you can't escape needing to interpolate.  The problem is that the template is being interpolated before being called by gettext(). So the variables are already filled in and gettext then can't find the translation. 

As a hack I tried `gettext('some ${var}')`

**Comment by dwaynebailey** [comment]
Right done a bit of reading.  I think we'll need to build a template string tag function.  These seem to work as such.

```js
l10n(string, ...values) {
   // Do stuff here to actually use gettext
   // Tehn interpolate
   return $string // Lastly return the now localised and interpolated string
}

l10n`Some ${var}`
```

**Comment by dwaynebailey** [comment]
@julen are you able to help in terms of getting template string working in PO?  This feels a bit beyond my skills in this area.

**Comment by dwaynebailey** [comment]
Upstream [xgettext bug](http://savannah.gnu.org/bugs/index.php?46111) as well as a [Mozilla related issue](https://github.com/mozilla/addons-linter/issues/50)

**Comment by unho** [comment]
It seems that @ta2-1 is providing a fix for this in #4903 so assigning him the issue.

**Comment by ta2-1** [comment]
Fixed in #4903.
